bgcval2/analysis_compare.py

```diff
@@ -3185,7 +3097,7 @@ def areatotal(nc, keys):
                     av[name]['modelFiles'])
                 if strictFileCheck: assert 0
 
-            if av[name]['dataFile'] != '':
+            if 'dataFile' in av[name].keys():
```

Comment:
```suggestion
            if 'dataFile' in av[name]:
```

Reply:
Careful here, av is an autovifiviation (because reasons), not a dict, so ` if 'dataFile' in av[name]` is always True.

Reply:
is a what?

Reply:
you mean a dict of dicts? - but then your conditional is too loose even with `.keys()` in it - `dict.keys()` is always a list, if looked at a simple dict, if looked at a dict of dicts, that'll return a `TypeError: unhashable type: 'dict'`

Reply:
It's a dict of dict, but you can create layers all at once. ie:
```
a = {}
a[1][2][3] = 4# Fails
```
Results in a failure.

But
```
a = autovivification()
a[1][2][3] = 4 # works find
```

Reply:
right, that's called a nested dict - you can initialize it like any other dict with `av = dict()` and then when you call `av.keys()` it'll return the main keys, and when you call `av[key].keys()` it'll return the keys of the `key` dict - but you can get rid of `keys()` in both places because `av` behaves like a dict and so does `av[key]`

Reply:
`dict.keys()` is always a list of the keys of `dict` - and it's always better to check for the existence of some `key` in that list rather than explicitly getting the `keys()` method out. Anyways, semantics - not breaking the shop, am not that anal :grin: